### DISCUSSION TITLE
ComponentFactory vs Dependency Injection

### DISCUSSION THREAD

**Discussion by SkyeHoefling** [dnnsoftware/Dnn.Platform#3951] [General] [answered]
The `ComponentFactory` is an implementation of the Factory Pattern that has been available in DNN for quite a long time. Has anyone looked at how this will grow with DNN Dependency Injection? I would argue that the `ComponentFactory` should be deprecated and replaced with Dependency Injection as most if not all of the features in the factory are available in a more streamlined approach with Dependency Injeciton.

I took a look through the GitHub issues and didn't see anything documented for this. Any change to the `ComponentFactory` will need to be a delicate change that should follow current deprecation guidelines. This would be  large breaking change for enterprise solutions

**Comment by bdukes** [comment, marked as answer]
I don't think anyone's documented this yet.  I had wondered if we could implement the `ComponentFactory` API on top of the DI container, but haven't gone down the road of trying any of that.

**Reply by SkyeHoefling** [reply]
I haven't tried a line of code yet on this, but I think we can safely swap out the `ComponentFactory` implementation to use DI. All the `ComponentFactory` is map types and implementations and hold it in memory when the DNN App starts up. This is why I think it would be safe to swap the implementation and then we could deprecate the `ComponentFactory`.

I'll write up an issue on this so we can get it documented

**Reply by SkyeHoefling** [reply]
Added #3955 to the issue backlog
